nncf/dynamic_graph/input_wrapping.py

```diff
@@ -0,0 +1,121 @@
+from collections import OrderedDict
+from inspect import Signature, Parameter
+from typing import List
+
+import torch
+
+from nncf.dynamic_graph.patch_pytorch import register_operator
+from nncf.dynamic_graph.graph_builder import ModelInputInfo, create_mock_tensor
+from nncf.utils import is_tensor, objwalk
+from nncf.nncf_logger import logger as nncf_logger
+
+
+@register_operator()
+def nncf_model_input(tensor: 'torch.Tensor'):
+    return tensor
+
+
+def wrap_nncf_model_inputs_helper(model_args, model_kwargs, input_infos: List['ModelInputInfo'] = None,
+                                  force_objwalk=False):
+    """Will wrap the input tensors among the model_args and model_kwargs with a no-op required for
+    correct input tracing by NNCF. If input_infos is supplied and force_objwalk is False, will only
+    affect the tensor args/kwargs as specified in the input_infos (usually populated via a corresponding
```

Comment:
would be great to test this scenario. As far as I can see, there're tests with direct wrapping all inputs inside `wrap_inputs_fn` (GNMT, SR Small model), but I can't find checking the logic with input_infos.

Reply:
This scenario is actually not required any more - replaced this call with an objwalk-based wrapping in the default dummy forward creation. The branch without `wrap_inputs_fn` in `create_dummy_forward` won't be executed normally when used in `NNCFNetwork` - `NNCFNetwork` always sets the `wrap_inputs_fn` to what it uses in its own forward to ensure consistency.

Reply:
but there's still a branch with info for missing inputs
https://github.com/openvinotoolkit/nncf/pull/242/files#diff-b2eabe2503ddc6713cecbeb5ff0740babbf6ad0bf3aac57e80f3f36d98a79a14R76

Reply:
Added tests for wrapping sequence for a given forward function signature, actual call and input infos.